src/Compilers/CSharp/Portable/Lowering/StateMachineRewriter/SynthesizedStateMachineMethod.cs

```diff
@@ -9,19 +10,18 @@ namespace Microsoft.CodeAnalysis.CSharp
     /// <summary>
     /// State machine interface method implementation.
     /// </summary>
-    internal sealed class SynthesizedStateMachineMethod : SynthesizedImplementationMethod, ISynthesizedMethodBodyImplementationSymbol
+    internal abstract class SynthesizedStateMachineMethodBase : SynthesizedImplementationMethod, ISynthesizedMethodBodyImplementationSymbol
```

Comment:
@gafter, Please suggest a better names for the affected types and I will be happy to use them.